src/github/github.py

```diff
@@ -0,0 +1,188 @@
+    )
+    rate_limit_remaining = int(result.headers["x-ratelimit-remaining"])
+    if rate_limit_remaining < 50:
+        raise Exception("Less than 50 api calls remaining before being rate limited, please try again later")
+
+    return result
+
+async def get_user_by_username(
+        username: str
+) -> GithubUser:
+    """
+        Takes in a Github username and returns an instance of GithubUser.
+
+        May return an empty list if no such user was found.
+    """
+    result = await _github_request_get(f"https://api.github.com/users/{username}",
+                              os.environ.get("GITHUB_TOKEN"))
+    result_json = result.json()
+    return [GithubUser(user["login"], user["id"], user["name"]) for user in [result_json]]
+
+async def get_user_by_id(
+    uid: str
+) -> GithubUser:
+    """
+        Takes in a Github user id and returns an instance of GithubUser.
+
+        May return an empty list if no such user was found.
+    """
+    result = await _github_request_get(f"https://api.github.com/user/{uid}",
+                              os.environ.get("GITHUB_TOKEN"))
+    result_json = result.json()
+    return [GithubUser(user["login"], user["id"], user["name"]) for user in [result_json]]
+
+async def add_user_to_org(
+        org: str = github_org_name,
+        uid: str | None = None,
+        email: str | None = None
+) -> None:
+    if uid is None and email is None:
+        raise Exception("uid and username cannot both be empty")
+    result = None
+    # Arbitrarily prefer uid
+    if uid is not None and email is None:
+        result = await _github_request_post(f"https://api.github.com/orgs/{org}/invitations",
+                               os.environ.get("GITHUB_TOKEN"),
+                               dumps({"invitee_id":uid, "role":"direct_member"}))
+    # Assume at this point both exist, but use email
+    else:
+        result = await _github_request_post(f"https://api.github.com/orgs/{org}/invitations",
+                               os.environ.get("GITHUB_TOKEN"),
+                               dumps({"email":email, "role":"direct_member"}))
+    result_json = result.json()
+    # Logging here potentially?
+    if result.status_code != 201:
+        raise Exception(f"Status code {result.status_code} returned when attempting to add user to org: {result_json['message']}: {[error['message'] for error in result_json['errors']]}")
+
+async def delete_user_from_org(
+        username: str,
+        org: str = github_org_name
+) -> None:
+    if username is None:
+        raise Exception("Username cannot be empty")
+    result = await _github_request_delete(f"https://api.github.com/orgs/{org}/memberships/{username}",
+                                          os.environ.get("GITHUB_TOKEN"))
+    # Logging here potentially?
+    if result.status_code != 204:
+        raise Exception(f"Status code {result.status_code} returned when attempting to delete user {username} from organization {org}")
```

Comment:
I'm hoping to log all exceptions that bubble up from an endpoint when a function is called. However, if we plan to handle these kinds of errors, then I would prefer to not use exceptions & log at a higher level. It's probably fine for now, but we might want to refactor a lil depending on how implementation goes.

Reply:
Yea, I want to replace these exceptions with logging calls, they're just here so they're blatantly obvious that we can replace them with logging calls.

Reply:
yee!